benchmarks/core/hanoi.bril

```diff
@@ -0,0 +1,26 @@
+# Three-disk Tower of Hanoi puzzle.
+#
+# Moves are printed in order, one on each line, where a move `src dst` indicates
+# that the top disk from rod `src` should be moved to rod `dst`.
+
+@hanoi (disk: int, src: int, dst: int, spare: int) {
+  zero: int = const 0;
+  nonneg: bool = ge disk zero;
+  br nonneg .then .else;
+.then:
+  one: int = const 1;
+  above: int = sub disk one;
+  call @hanoi above src spare dst;
+  print src dst;
+  call @hanoi above spare dst src;
+.else:
+  ret;
+}
+
+@main {
+  disk: int = const 2;
+  src: int = const 0;
+  dst: int = const 2;
+  spare: int = const 1;
```

Comment:
FWIW, if you're interested, you could pass these as parameters from the test harness instead of hard-coding them in `@main`. That would (hypothetically) make it easier to run the benchmark on different inputs. Here's an example:
https://github.com/sampsyo/bril/blob/a7a0759b20e724260825fc9c6ae7273a09c0bafd/benchmarks/core/armstrong.bril#L1-L2

Reply:
Updated to take the number of disks as a parameter. The other constants are more or less baked in to the design of the game, so I left those hard coded.